modules/liquidity-pools/src/lib.rs

```diff
@@ -257,6 +299,27 @@ impl<T: Trait> SyntheticProtocolLiquidityPools<T::AccountId> for Module<T> {
 	}
 }
 
+impl<T: Trait> MarginProtocolLiquidityPools<T::AccountId> for Module<T> {
+	type TradingPair = TradingPair<T::CurrencyId>;
+	fn get_swap_rate(pool_id: Self::LiquidityPoolId, pair: Self::TradingPair) -> Fixed128 {
+		Self::swap_rate(pool_id, (pair.base, pair.quote))
+	}
+
+	fn get_accumulated_swap_rate(pool_id: Self::LiquidityPoolId, pair: Self::TradingPair) -> Fixed128 {
+		Self::accumulated_swap_rate(pool_id, (pair.base, pair.quote))
+	}
+
+	fn can_open_position(
+		pool_id: Self::LiquidityPoolId,
+		pair: Self::TradingPair,
+		leverage: Leverage,
+		_leveraged_amount: Self::Balance,
+	) -> bool {
+		// FIXME: this implementation may change
+		Self::is_enabled(pool_id, pair.base, leverage) && Self::is_enabled(pool_id, pair.quote, leverage)
```

Comment:
In my understanding, enabled synthetic protocol currencies doesn't necessarily mean any two of them are supported trading pairs in margin protocol. We might need the functionality for LPs to manage enabled margin trading pairs? @xlc

Reply:
@shaopengw I'm not sure either that's why added FIXME so we don't forget to update it